### DISCUSSION TITLE
Add sub page for tab

### DISCUSSION THREAD

**Discussion by PlexSheep** [cotes2020/jekyll-theme-chirpy#1661] [Q&A]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Built from `jekyll-theme-chirpy`

### Description

How do I add a sub page to my tab, so that I can have an URL like `http://example.com/demotab/page`?

### Operations you have already tried

- Creating a directory in `_tabs`
- Just adding a markdown link to a local markdown file

**Comment by lotusk08** [comment]
You can create a subfolder like this:
.
├── about.md          # => http://example.com/about.html
├── documentation     # folder containing pages
│   └── doc1.md       # => http://example.com/documentation/doc1.html
├── design            # folder containing pages
│   └── draft.md      # => http://example.com/design/draft.html

**Reply by PlexSheep** [reply]
Thanks, I'll try that later. I thought I did that already but Apperently I didn't give it enough effort.

**Reply by PlexSheep** [reply]
I'm probably doing something wrong, but I still can't get it to work. My tabs are like this now:
```
_tabs
├── about.md
├── archives.md
├── categories.md
├── imprint.md
├── links.md
├── security.md
├── tags.md
└── test
    ├── a.md
    └── b.md

2 directories, 9 files
```

The `test` tab does not appear in the sidebar and the link <http://localhost:4000/test/a.html> does not work

**Comment by kungfux** [comment]
Here are a few suggestions, so you can give it a try.

### 1. Custom URL
You can set the desired custom URL for a page as following:

```markdown
---
# the default layout is 'page'
# layout: page
permalink: /demotab/page/
---
```

The page may be resolved as:
- `/demotab/page/`
- `/demotab/page/index.html`

If you want, you can specify the full exact path like `/demotab/page.html` instead.
Read more here: [1](https://jekyllrb.com/docs/front-matter/#predefined-global-variables), [2](https://jekyllrb.com/docs/permalinks/).

### 2. Separate folder

However, this page will still appear in the list of tabs on the sidebar. If you don't want to see it there, please try to create a new folder outside the `_tabs` and place your `.md` pages there e.g. `./demotab/page.md`. This may require to specify layout and maybe other things to make it work correctly.

### 3. Collections

Frankly, I know nothing about collections, so propose you to read about it in [documentation](https://jekyllrb.com/docs/collections/) just in case you find it useful or if nothing else works for you.

Good luck.
